src/dlgIRC.cpp

```diff
@@ -29,22 +31,49 @@
 #include <QDesktopServices>
 #include <QDir>
 #include <QScrollBar>
+#include <QShortcut>
 #include <QString>
+#include <QTextDocument>
 #include <QTime>
 #include "post_guard.h"
 
 
 dlgIRC::dlgIRC()
 {
+    mInputHistoryMax = 8;
+
     setupUi(this);
-    connection = new IrcConnection(this);
-    irc->setOpenExternalLinks(true);
-    setUnifiedTitleAndToolBarOnMac(true);
-    connect(irc, SIGNAL(anchorClicked(QUrl)), this, SLOT(anchorClicked(QUrl)));
-    connect(connection, SIGNAL(messageReceived(IrcMessage*)), this, SLOT(onMessageReceived(IrcMessage*)));
-    connect(connection, SIGNAL(connected()), this, SLOT(onConnected()));
-    connect(lineEdit, SIGNAL(returnPressed()), this, SLOT(sendMsg()));
+    setupCommandParser();
 
+    connection = new IrcConnection(this);
+    connection->setReconnectDelay(5);
+
+    lineEdit->setAttribute(Qt::WA_MacShowFocusRect, false);
+    ircBrowser->setFocusProxy( lineEdit );
+
+    // nick name completion
+    completer = new IrcCompleter(this);
+    completer->setParser(commandParser);
+    connect(completer, SIGNAL(completed(QString,int)), this, SLOT(slot_nameCompleted(QString,int)));
+    QShortcut* shortcut = new QShortcut(Qt::Key_Tab, this);
+    connect(shortcut, SIGNAL(activated()), this, SLOT(slot_nameCompletion()));
+    QShortcut* shortcut2 = new QShortcut(Qt::Key_Up, this);
+    connect(shortcut2, SIGNAL(activated()), this, SLOT(slot_onHistoryCompletion()));
+
+    connect(lineEdit, SIGNAL(returnPressed()), this, SLOT(slot_onTextEntered()));
```

Comment:
Did you know you could use compile-time signal connections? That way any errors are discovered when you're compiling and not at runtime - it's safer. I think if you right-click on this in Qt Creator and select refactor, it'll give you the option to convert.